src/gpio.rs

```diff
@@ -457,6 +475,24 @@ where
         !self.is_set_high()
     }
 
+    /// Returns the level to which this pin is currently set
+    ///
+    /// This method is only available, if two conditions are met:
+    /// - The pin is in the GPIO state.
+    /// - The pin direction is set to output.
+    ///
+    /// See [`Pin::into_output_pin`] and [`into_output`]. Unless both of these
+    /// conditions are met, code trying to call this method will not compile.
+    ///
+    /// [`Pin::into_output_pin`]: ../pins/struct.Pin.html#method.into_output_pin
+    /// [`into_output`]: #method.into_output
+    pub fn get_set_level(&self) -> Level {
+        match self.is_set_high() {
+            true => Level::High,
+            false => Level::Low,
+        }
```

Comment:
Suggestion: To reduce code duplication, how about an `impl From<bool> for Level`?

Reply:
Hrrrrm. Sounds enticing at first (declutter! yay!) but from my perspective as an outsider getting to know the codebase, I think this is a potential cause of confusion:
We'd be ascribing semantics to a boolean which aren't intuitively clear if you don't know the backstory:
"Does `level == true` mean it's high because there's current running through it? Or that it's low because as a newcomer, I'm used to "pin low -> LED on"? Oh it's true if `is_high()` is true? Well, I didn't expect *that*"

This could and should of course be documented, but it's still a quirk, and I don't know if it helps with readability even if it removes clutter.

Reply:
Hmm, yeah, I see what you mean. So how about an inherent method instead? `Level::from_high(bool)`? `Level::from_is_set_high(bool)`? Should reduce duplication all the same, while alleviating your concern.

Reply:
Or even `Level::read(&GpioPin)`?

Reply:
check! implemented `Level::read(&GpioPin)`, but called it `Level:: from_pin()` :)